### PR TITLE
expedition rewards

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
requires changes to cargo system to be able to add and approve an item rather than a cargoProduct

Requires https://github.com/space-wizards/RobustToolbox/pull/4142

**Media**
rewards added upon ftling back, spawner needs to have sprite changed or something but yeah:

spawner still spawns stuff:

spawners use actual icons

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: NanoTrasen has finally started rewarding salvagers for actually completing their missions. Rewards can range from materials, to cloning parts, to advanced weaponry.
- remove: todo...

### CODE REVIEW

**Comment by deltanedas** [comment]
@metalgearsloth besides already mentioned cargo concerns, how would i get the difficulty rating for the expedition component

**Comment by metalgearsloth** [comment]
> @metalgearsloth besides already mentioned cargo concerns, how would i get the difficulty rating for the expedition component

You can probably put the difficulty on the comp itself if it's not there (rn you could just derive it from the seed but selectable difficulty means it'd need to be there).

**Comment by metalgearsloth** [comment]
Add in cloner plz.

I think the amount of reward for 20 minutes work is probably too low, maybe make it spawn like 3 crates? Maybe like 3 common on normal, 1 common 2 rare on hard, 1 rare 2 epic on hardets or smth. I'd say 5 but if I'm leaning towards allowing multiple ops at once.

File: Content.Server/Salvage/SalvageSystem.Runner.cs [hunk 1]
```diff
@@ -272,8 +275,36 @@ private void UpdateRunner()
...
+        while (rewardQuery.MoveNext(out var uid, out var comp))
+        {
+            if (comp.Completed)
+            {
                 Announce(uid, Loc.GetString("salvage-expedition-completed"));
+                GiveReward(uid, comp);
             }
         }
     }
+
```
Comment (metalgearsloth) [inline Content.Server/Salvage/SalvageSystem.Runner.cs:292]: Isn't this going to spam the reward every tick?

Comment (deltanedas) [reply Content.Server/Salvage/SalvageSystem.Runner.cs:292]: isnt it deleted after

**Comment by deltanedas** [comment]
@metalgearsloth should the 3 rewards of different tiers be made apparent in mission ui or should that be left for when choosing mission difficulty is implemented

**Comment by deltanedas** [comment]
@metalgearsloth ok rewards work (with inlined Frame0 to run duh) i got more questions:
- should i change the spawners sprite to be a t3/t4 part respectively so it looks correct in the cargo console or is there a smarter way that still looks like a spawner when mapping stuff
- should i remove random loot spawners from maps to not get 50 stacks of mats and not even complete the mission, or just make them rarer
- should i remove the laser cannon xray cannon etc from maps now that they are rewards

**Comment by metalgearsloth** [comment]
> should i change the spawners sprite to be a t3/t4 part respectively so it looks correct in the cargo console or is there a smarter way that still looks like a spawner when mapping stuff
Uhhh yeah if it's not used anywhere.

> should i remove random loot spawners from maps to not get 50 stacks of mats and not even complete the mission, or just make them rarer

Steel sheets / canisters are fine to be mapped in I think, just anything rarer than that (e.g. xrays, lasers, ATV, APE) probably remove.

**Comment by metalgearsloth** [comment]
So my general idea long-term:

- Explicit mission rewards are station shit like T3 parts or cloner or whatever.
- Missions themselves have either mining or common materials spawned, nothing rare, like basic circuitboards or whatever. At best it'd be something like a PKA upgrade or the likes from opposing salvagers.
- Expedition mining + shaft mining gives bulk minerals.
- Expeditions also have chemplants that can be ground up for when we remove the infinite chem dispenser but need sprites for this. Same with potential NPC critters.
- Salvage magnet pulls in wrecks you can deconstruct for money (using a similar UI to expeditions) and has no combat whatsoever and is procedurally generated (you'd pretty much re-use the dungeon code for this), although this stuff sells at a reduced rate. Think like hardspace shipbreakers.
- Shaft mining would probably just be nearby asteroids that are essentially what the cave mining missions for expeditions are like with very sporadic mobs.

**Comment by deltanedas** [comment]
for last 2 basically afterlight :trollface:

**Comment by metalgearsloth** [comment]
The last one would probably be using AL's worldgen though I have NFI how to use it.

**Comment by deltanedas** [comment]
copy paste le asteroid generator but instead of procedurally for the entire map for a single position

**Comment by metalgearsloth** [comment]
It's already in the repo but it has a billion options and its own queuedel implementation and scary.

**Comment by deltanedas** [comment]
ok so:
- removed all mats and mat spawners
- removed akms gatfruit secfab lasers drozd revolvers and mags
- added some plushies and books in place of gold/silver/secfab

should i add gatfruit back as a high tier reward since funny

**Comment by metalgearsloth** [comment]
> should i add gatfruit back as a high tier reward since funny

Sure

**Comment by deltanedas** [comment]
,pro says after merging it :trollface:

**Pull request merged**